src/components/Provider.svelte

```diff
@@ -28,33 +37,54 @@
   transactionSigningAddress.subscribe(async (val) => {
     localSigningAddress = val;
     if (api) {
-      accountInfo = await getBalances(api, val);
+      accountBalances = await getBalances(api, val);
     }
   });
 
+  // the locally stored value of the provider Id
+  let msaId = 0;
+  let isProvider = false;
+
+  storeMsaInfo.subscribe((info: MsaInfo) => {
+    msaId = info?.msaId || 0;
+    isProvider = info?.isProvider || false
+  });
+
   const balanceToHuman = (balance: bigint): string => {
     return formatBalance(balance, { withSiFull: true, withUnit: token, withZero: true });
   };
 
   function showAddControlKey() {
-    storeCurrentAction.update((val) => (val = ActionForms.AddControlKey));
+    storeCurrentAction.set(ActionForms.AddControlKey);
```

Comment:
Should this `.set()` be updated for `showStake()` as well?

Reply:
doesn't really make a difference; it's simply more legible.